core/consensus/babe/src/lib.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! BABE (Blind Assignment for Blockchain Extension) consensus in substrate.
+#![forbid(warnings, unsafe_code)]
+extern crate parity_codec_derive;
+use babe_primitives::BABE_ENGINE_ID;
+use parity_codec::{Decode, Encode, Input};
+use runtime_primitives::generic;
+use schnorrkel::{
+    sign::Signature,
+    vrf::{VRFProof, VRF_PROOF_LENGTH},
+    PublicKey, SecretKey, PUBLIC_KEY_LENGTH, SIGNATURE_LENGTH,
+};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct BabeSealSignature {
+    public_key: PublicKey,
```

Comment:
I would advocate using the `substrate-primitives::sr25519::PublicKey` instead -- this is a wrapper around schnorrkel functionality (just add it if you need more)

Reply:
Agreed.